File: app/src/main/java/com/davismiyashiro/expenses/view/BaseCompatActivity.java
package com.davismiyashiro.expenses.view;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.Menu;
import android.view.MenuItem;

import com.davismiyashiro.expenses.R;

import timber.log.Timber;

public abstract class BaseCompatActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        Timber.d("onCreate() event");
        super.onCreate(savedInstanceState);

        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
    }

    @Override
    public void onStart() {
        super.onStart();
        Timber.d("onStart() event");
    }

    @Override
    public void onRestart() {
        super.onRestart();
        Timber.d("onRestart() event");
    }

    @Override
    public void onResume() {
        super.onResume();
        Timber.d("onResume() event");
    }

    @Override
    public void onPause() {
        super.onPause();
        Timber.d("onPause() event");
    }

    @Override
    public void onStop() {
        super.onStop();
        Timber.d("onStop() event");
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        Timber.d("onDestroy() event");
    }
}

File: app/src/main/java/com/davismiyashiro/expenses/view/BaseCompatActivity.kt
/*
 * MIT License
 *
 * Copyright (c) 2019 Davis Miyashiro
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
package com.davismiyashiro.expenses.view

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.Toolbar

import com.davismiyashiro.expenses.R

import timber.log.Timber

abstract class BaseCompatActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        Timber.d("onCreate() event")
        super.onCreate(savedInstanceState)

        val toolbar: Toolbar = findViewById(R.id.toolbar)
        setSupportActionBar(toolbar)
    }

    public override fun onStart() {
        super.onStart()
        Timber.d("onStart() event")
    }

    public override fun onRestart() {
        super.onRestart()
        Timber.d("onRestart() event")
    }

    public override fun onResume() {
        super.onResume()
        Timber.d("onResume() event")
    }

    public override fun onPause() {
        super.onPause()
        Timber.d("onPause() event")
    }

    public override fun onStop() {
        super.onStop()
        Timber.d("onStop() event")
    }

    public override fun onDestroy() {
        super.onDestroy()
        Timber.d("onDestroy() event")
    }
}
